src/contracts/GPv2Settlement.sol

```diff
@@ -342,21 +348,32 @@ contract GPv2Settlement is GPv2Signing, ReentrancyGuard, StorageAccessible {
         }
     }
 
-    /// @dev Claims order gas refunds by freeing storage for all encoded order
-    /// gas refunds.
+    /// @dev Claims order gas refunds.
+    ///
+    /// @param orderRefunds Order refund data for freeing storage.
+    function claimOrderRefunds(OrderRefunds calldata orderRefunds) internal {
+        freeOrderStorage(orderRefunds.filledAmounts, filledAmount);
+        freeOrderStorage(orderRefunds.preSignatures, preSignature);
```

Comment:
With this approach an order smart contract order would have to be included in both filledAmounts as well as in preSignatures in order to refund it. Would it be safe to assume that for all preSignature removals we also remove filled amounts?

Reply:
It is possible for an order to get signed and not filled, which is why I kept them separate.